analyzers/src/SonarAnalyzer.CSharp/Rules/ClearTextProtocolsAreSensitive.cs

```diff
@@ -0,0 +1,165 @@
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(EnableSslRule, objectCreation.GetLocation()));
+            }
+        }
+
+        private void VisitInvocationExpression(SyntaxNodeAnalysisContext context)
+        {
+            var invocation = (InvocationExpressionSyntax)context.Node;
+            if (telnetRegexForIdentifier.IsMatch(invocation.Expression.ToString()))
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(DefaultRule, invocation.GetLocation(), TelnetKey, recommendedProtocols[TelnetKey]));
+            }
+        }
+
+        private void VisitAssignments(SyntaxNodeAnalysisContext context)
+        {
+            var assignment = (AssignmentExpressionSyntax)context.Node;
+            if (assignment.Left is MemberAccessExpressionSyntax memberAccess
+                && memberAccess.IsMemberAccessOnKnownType(EnableSslName, KnownType.System_Net_FtpWebRequest, context.SemanticModel)
+                && context.SemanticModel.GetConstantValue(assignment.Right) is { HasValue: true, Value: bool enableSslValue }
+                && !enableSslValue)
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(EnableSslRule, assignment.GetLocation()));
+            }
+        }
+
+        private void VisitStringExpressions(SyntaxNodeAnalysisContext c)
+        {
+            var text = GetText(c.Node);
+            if (GetUnsafeProtocol(text) is {} unsafeProtocol)
+            {
+                c.ReportDiagnosticWhenActive(Diagnostic.Create(DefaultRule, c.Node.GetLocation(), unsafeProtocol, recommendedProtocols[unsafeProtocol]));
+            }
+        }
+
+        private bool IsServerSafe(ObjectCreationExpressionSyntax objectCreation) =>
+            objectCreation.ArgumentList?.Arguments.Count > 0
+            && validServerRegex.IsMatch(GetText(objectCreation.ArgumentList.Arguments[0].Expression));
+
+        private string GetUnsafeProtocol(string text)
+        {
+            if (httpRegex.IsMatch(text))
+            {
+                return "http";
+            }
+            else if (ftpRegex.IsMatch(text))
+            {
+                return "ftp";
+            }
+            else if (telnetRegex.IsMatch(text))
+            {
+                return "telnet";
+            }
+
+            return null;
+        }
+
+        private static string GetText(SyntaxNode node) =>
+            node switch
+            {
+                InterpolatedStringExpressionSyntax interpolatedStringExpression => interpolatedStringExpression.GetContentsText(),
+                LiteralExpressionSyntax literalExpression => literalExpression.Token.ValueText,
+                _ => string.Empty
+            };
+
+        private static Regex CreateCompiledRegex(string pattern) =>
+            new Regex(pattern, RegexOptions.Singleline | RegexOptions.Compiled);
```

Comment:
Can you either remove the SingleLine or add a test case demonstrating multiline string where the `.` matches the new line character?
```suggestion
            new Regex(pattern, RegexOptions.Compiled);
```